### PR TITLE
discussions widget and notifications group setting

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: mod/discussions/views/default/widgets/discussions/content.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+$owner = $widget->getOwnerEntity();
+$url = elgg_generate_url('collection:object:discussion:all');
+
+if ($owner instanceof \ElggUser) {
+	if ($widget->context === 'dashboard') {
+		if (elgg_get_plugin_setting('enable_global_discussions', 'discussions') !== 1) {
+			$groups = $owner->getGroups([
+				'limit' => false,
+				'batch' => true,
+			]);
```
Comment (jeabakker) [inline mod/discussions/views/default/widgets/discussions/content.php:31]: use callback to get all the group_guids at once, no batch needed

File: mod/discussions/views/default/widgets/discussions/content.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		$url = elgg_generate_url('collection:object:discussion:owner', ['username' => $owner->username]);
+	}
+} elseif ($owner instanceof \ElggGroup) {
+	$options['container_guid'] = $owner->owner_guid;
+	$url = elgg_generate_url('collection:object:discussion:group', ['guid' => $owner->guid]);
+}
+
+$content = elgg_list_entities($options);
+if (empty($content)) {
+	echo elgg_echo('discussion:none');
```
Comment (jeabakker) [inline mod/discussions/views/default/widgets/discussions/content.php:54]: use `page/components/no_results`

File: mod/discussions/views/default/discussion/groups/settings.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	'#type' => 'checkbox',
+	'#label' => elgg_echo('discussions:groups:edit:add_group_subscribers_to_discussion_comments'),
+	'name' => 'settings[discussions][add_group_subscribers_to_discussion_comments]',
+	'checked' => $checked,
+	'switch' => true,
+	'default' => 0,
+	'value' => 1,
+]);
+
+echo elgg_view_module('info', elgg_echo('collection:object:discussion'), $content);
```
Comment (jeabakker) [inline mod/discussions/views/default/discussion/groups/settings.php:23]: enter ;)

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: mod/discussions/views/default/widgets/discussions/content.php (hunk 2 above)
Comment (jdalsem) [reply mod/discussions/views/default/widgets/discussions/content.php:54]: keeping this inline with other content plugins... reasons is probably excessive whitespace issues if using the page component

**Review by jeabakker — APPROVED** [review]

**Review by jeabakker — APPROVED** [review]

**Pull request merged**